sample-code/spring-app/src/test/java/com/sap/ai/sdk/app/services/OpenAiServiceV2.java

```diff
@@ -147,54 +139,33 @@ public CreateChatCompletionResponse chatCompletionToolExecution(
             .name("weather")
             .description("Get the weather for the given location")
             .parameters(schemaMap);
-
     final var tool = new ChatCompletionTool().type(FUNCTION).function(function);
 
-    var userMessage =
-        new ChatCompletionRequestUserMessage()
-            .role(ChatCompletionRequestUserMessage.RoleEnum.USER)
-            .content(
-                ChatCompletionRequestUserMessageContent.create(
-                    "What's the weather in %s in %s?".formatted(location, unit)));
+    final var messages = new ArrayList<OpenAiMessage>();
+    messages.add(OpenAiMessage.user("What's the weather in %s in %s?".formatted(location, unit)));
 
-    final var request =
-        new CreateChatCompletionRequest()
-            .addMessagesItem(userMessage)
-            .tools(List.of(tool))
-            .functions(null);
+    final var request = new OpenAiChatCompletionRequest(messages).withTools(List.of(tool));
+    var response = OpenAiClient.forModel(GPT_4O_MINI).chatCompletion(request);
 
-    OpenAiClient client = OpenAiClient.forModel(GPT_4O_MINI);
+    final var assistantMessage = response.getMessage();
+    messages.add(assistantMessage);
 
-    final var initialResponse = client.chatCompletion(request);
+    final var functionCall = (OpenAiFunctionCallItem) assistantMessage.getToolCalls().get(0);
 
-    final var toolCall = initialResponse.getChoices().get(0).getMessage().getToolCalls().get(0);
     String toolResponseJson;
     try {
       var weatherRequest =
-          JACKSON.readValue(toolCall.getFunction().getArguments(), WeatherMethod.Request.class);
+          JACKSON.readValue(functionCall.getArguments(), WeatherMethod.Request.class);
       final var toolResponse = new WeatherMethod().getCurrentWeather(weatherRequest);
       toolResponseJson = JACKSON.writeValueAsString(toolResponse);
     } catch (JsonProcessingException e) {
       throw new IllegalArgumentException("Error parsing tool call arguments", e);
     }
 
-    final var assistantMessage =
-        new ChatCompletionRequestAssistantMessage()
-            .role(ChatCompletionRequestAssistantMessage.RoleEnum.ASSISTANT)
-            .content(
-                ChatCompletionRequestAssistantMessageContent.create(
-                    initialResponse.getChoices().get(0).getMessage().getContent()))
-            .toolCalls(List.of(toolCall));
-    request.addMessagesItem(assistantMessage);
-
-    final var toolMessage =
-        new ChatCompletionRequestToolMessage()
-            .role(ChatCompletionRequestToolMessage.RoleEnum.TOOL)
-            .content(ChatCompletionRequestToolMessageContent.create(toolResponseJson))
-            .toolCallId(toolCall.getId());
-    request.addMessagesItem(toolMessage);
-
-    return client.chatCompletion(request);
+    final var toolMessage = OpenAiMessage.tool(toolResponseJson, functionCall.getId());
```

Comment:
The Javadoc on both `OpenAiToolMessage` and `OpenAiMessage` talks about "message" instead of "tool call execution response in the form of JSON"